docs/codegen/overview.md

```diff
@@ -55,7 +93,8 @@ generate_file "sum.txt" {
 
 # Assertions
 
-Assertions can be used in order to fail code generation for one or more stacks
+Assertions can be only used when the [generation context](#generation-context) is
```

Comment:
Hmm very subjective, so feel free to disregard, but it seems to me that it would be better to explain assertions and then after the explanation mention that they are limited to context=stack (instead of it being the first piece of information).

Reply:
fixed